### PR TITLE
Pulling driver specs out of configuruation.md

### LINKED ISSUE #1254: Redundancy / unwieldy configuration / storage docs
Issue 1: configuration reference is growing really big

Issue 2: drivers separate docs duplicate information, leading to inconsistencies and increased maintenance cost

Example:

https://github.com/docker/distribution/blob/master/docs/configuration.md#openstack-swift

https://github.com/docker/distribution/blob/master/docs/storage-drivers/swift.md

Suggestions / questions:

 * is there a point in maintaining separate drivers documents?
 * or: can't we link the drivers documents from the reference?

cc @moxiegirl @donhcd @RichardScothern @BrianBland and anyone else on the docs team interested

### PR DESCRIPTION
**Pull request by moxiegirl**
Fixes #1254 

Signed-off-by: Mary Anthony <mary@docker.com>

### CODE REVIEW

**Comment by moxiegirl** [comment]
@RichardScothern @stevvooe @dmp42

File: docs/configuration.md [hunk 1]
```diff
@@ -364,6 +376,18 @@ Permitted values are `error`, `warn`, `info` and `debug`. The default is
...
+        accesskeysecret: accesskeysecret
+        region: OSS region name
+        endpoint: optional endpoints
+        internal: optional internal endpoint
+        bucket: OSS bucket
+        encrypt: optional data encryption setting
+        secure: optional ssl setting
+        chunksize: optional size valye
+        rootdirectory: optional root directory
+      inmemory:
```
Comment (stevvooe) [inline docs/configuration.md]: Maybe a comment: "this driver takes no parameters".

File: docs/configuration.md [hunk 2]
```diff
@@ -378,9 +402,33 @@ Permitted values are `error`, `warn`, `info` and `debug`. The default is
...
+
+* The [`s3` storage backend](storage-drivers/s3.md) uses Amazon's Simple Storage Service (S3).
 
-If you are deploying a registry on Windows, be aware that a Windows volume mounted from the host is not recommended. Instead, you can use a S3, or Azure, backing data-store. If you do use a Windows volume, you must ensure that the `PATH` to the mount point is within Windows' `MAX_PATH` limits (typically 255 characters). Failure to do so can result in the following error message:
+* The [`swift` storage backend](storage-drivers/swift.md) uses Openstack Swift object storage.
+
+* The [`oss` storage backend](storage-drivers/oss.md) uses Openstack Swift object storage.
+
+For purely tests purposes, you can use the [`inmemory` storage
+driver](storage-drivers/inmemory.md). **Never** use this option in a production
```
Comment (stevvooe) [inline docs/configuration.md]: This might be a little strong. This is not as bad as, say, silly auth. "This driver is not intended for production usage. If you would like to run a registry from volatile memory, use the `filesystem` driver on a ramdisk."

File: docs/configuration.md [hunk 3]
```diff
@@ -378,9 +402,33 @@ Permitted values are `error`, `warn`, `info` and `debug`. The default is
         disable: false
 
 The storage option is **required** and defines which storage backend is in use.
-You must configure one backend; if you configure more, the registry returns an error.
+You must configure one backend; if you configure more, the registry returns an error. You can choose any of these backend storage drivers:
+
+* The [`filesystem` storage backend](storage-drivers/filesystem.md) uses the local
```
Comment (stevvooe) [inline docs/configuration.md]: I feel like this would be better as a table with name, link to docs, description and caveats.

File: docs/configuration.md [hunk 4]
```diff
@@ -378,9 +402,33 @@ Permitted values are `error`, `warn`, `info` and `debug`. The default is
...
+* The [`gcs` storage backend](storage-drivers/gcs.md) uses Google Cloud Storage.
+
+* The [`rados` storage driver](storage-drivers/rados.md) uses Ceph Object Storage.
+
+* The [`s3` storage backend](storage-drivers/s3.md) uses Amazon's Simple Storage Service (S3).
 
-If you are deploying a registry on Windows, be aware that a Windows volume mounted from the host is not recommended. Instead, you can use a S3, or Azure, backing data-store. If you do use a Windows volume, you must ensure that the `PATH` to the mount point is within Windows' `MAX_PATH` limits (typically 255 characters). Failure to do so can result in the following error message:
+* The [`swift` storage backend](storage-drivers/swift.md) uses Openstack Swift object storage.
+
+* The [`oss` storage backend](storage-drivers/oss.md) uses Openstack Swift object storage.
```
Comment (stevvooe) [inline docs/configuration.md]: This is Aliyun OSS.

File: docs/configuration.md [hunk 5]
```diff
@@ -364,6 +376,18 @@ Permitted values are `error`, `warn`, `info` and `debug`. The default is
         region: fr
         container: containername
         rootdirectory: /swift/object/name/prefix
+      oss:
```
Comment (stevvooe) [inline docs/configuration.md]: Maybe this belongs in another PR, but it seems like we should push these examples into the storage driver specific documentation.

File: docs/configuration.md [hunk 6]
```diff
@@ -1035,7 +630,7 @@ For more information about Token based authentication configuration, see the [sp
 ### htpasswd
 
 The _htpasswd_ authentication backed allows one to configure basic auth using an
-[Apache HTPasswd File](https://httpd.apache.org/docs/2.4/programs/htpasswd.html).
+[Apache HTPasswdFile](https://httpd.apache.org/docs/2.4/programs/htpasswd.html).
```
Comment (dmp42) [inline docs/configuration.md]: I think I would keep that separated - even downcase.

File: docs/storage-drivers/inmemory.md [hunk 7]
```diff
@@ -7,11 +7,14 @@ keywords = ["registry, service, driver, images, storage,  in-memory"]
...
 
 
-# In-memory storage driver
+# In-memory storage driver (Testing Only)
 
-An implementation of the `storagedriver.StorageDriver` interface which uses local memory for object storage.
+The `inmemory` storage driver is useful for testing purposes. This driver is an
+implementation of the `storagedriver.StorageDriver` interface which uses local
+memory for object storage.  If you use this driver, you should use ramdisk on
+your filesystem over pure in-memory.
```
Comment (dmp42) [inline docs/storage-drivers/inmemory.md]: I guess @stevvooe remark from above applies below as well.
Also, maybe this sentence can be enhanced?

File: docs/configuration.md (hunk 5 above)
Comment (moxiegirl) [inline docs/configuration.md]: Including the entire file and its options is pretty standard.  It lets readers compare and contrast elements and features.  If we didn't do this, we force multiple clicks to discover.

**Comment by moxiegirl** [comment]
@stevvooe @dmp42 Comments entered.  Let me know if I mis-interpreted or missed anything.

**Comment by moxiegirl** [comment]
@RichardScothern Can you re-run this CircleCI test? I don't have the perms in CircleCI to do that with this repo.

**Comment by moxiegirl** [comment]
@dmp42 @RichardScothern @stevvooe I don't have perms to merge so y'all have to move it forward it you want it.

File: docs/configuration.md [hunk 8]
```diff
@@ -446,7 +524,7 @@ The `redirect` subsection provides configuration for managing redirects from
 content backends. For backends that support it, redirecting is enabled by
 default. Certain deployment scenarios may prefer to route all data through the
 Registry, rather than redirecting to the backend. This may be more efficient
-when using a backend that is not colocated or when a registry instance is
+when using a backend that is not collocated or when a registry instance is
```
Comment (stevvooe) [inline docs/configuration.md]: ["Colocated"](http://www.merriam-webster.com/dictionary/colocate) is the correct word. ["collocated"](http://www.merriam-webster.com/dictionary/collocate) has to due with word placement. "Co-located" may also be acceptable here.

File: docs/configuration.md (hunk 1 above)
Comment (stevvooe) [inline docs/configuration.md]: @moxiegirl bump.

**Comment by moxiegirl** [comment]
@stevvooe sorted. merge as you please.

**Comment by stevvooe** [comment]
@moxiegirl This one? https://github.com/docker/distribution/pull/1341/files#r49544606

File: docs/configuration.md (hunk 1 above)
Comment (moxiegirl) [inline docs/configuration.md]: Hmm I see the comment in the source. GitHub seems to be hiding it in this diff.  You can see it on the page here:

https://github.com/moxiegirl/distribution/blob/fix-issue-1254/docs/configuration.md

It also shows fine in the built docs:

image

Comment (stevvooe) [inline docs/configuration.md]: Odd. I think there is a caching issue in github.

**Comment by stevvooe** [comment]
LGTM

**Comment by moxiegirl** [comment]
@dmp42 @RichardScothern would one of you like to do a review and merge?  Ty.

**Comment by dmp42** [comment]
LGTM
